pkg/provider/azure_instances.go

```diff
@@ -253,9 +254,14 @@ func (az *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID st
 
 		return false, err
 	}
-	klog.V(5).Infof("InstanceShutdownByProviderID gets power status %q for node %q", powerStatus, nodeName)
+	klog.V(5).Infof("InstanceShutdownByProviderID gets power status %q and provisioning state %q for node %q", powerStatus, provisioningState, nodeName)
 
 	status := strings.ToLower(powerStatus)
+	if status == vmPowerStateStopped && strings.EqualFold(strings.ToLower(provisioningState), strings.ToLower(string(compute.ProvisioningStateCreating))) {
```

Comment:
I don't think we should be doing/logging an "assumption". This is addressing a symptom rather than the cause.

I'd prefer if we the code is changed to:

```
return provisioningState == Succeeded && (status == vmPowerStateStopped || status == vmPowerStateDeallocated || status == vmPowerStateDeallocating)
```

Reply:
to reword my point: we should only act on **terminal states**  and avoid acting on transient states because they are more susceptible to latencies/API caching.

Reply:
because of caching, there're high chances the transient states are cached for instances. waiting another 10 minutes for caching refreshing might be too long

Reply:
We only care about the shutdown state for the sake of adding the annotation right? We aren't modifying the cache state anywhere with that change. This change is just for returning the result to `InstancesShutDown` call which shouldn't affect anything else apart from the taint.

Looking at the controller-manager code, once the node reports Ready - the taint will be cleared up by that control loop independent of the cached state in cloudprovider.

Reply:
makes sense. changed according to above suggestions